libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -604,9 +604,8 @@ class MapboxNavigation(
     private fun createInternalOffRouteObserver() = object : OffRouteObserver {
         override fun onOffRouteStateChanged(offRoute: Boolean) {
             if (offRoute) {
-                reroute()
-            } else {
                 rerouteController?.interrupt()
+                reroute()
```

Comment:
seems that in case when we/customer call `reroute` we alway must interrupt current reroute call, so I'd suggest to move interrupt under `reroute` in `MapboxRerouteController`.
So `MapboxRerouteController.kt` will look like
```
private fun request(routeOptions: RouteOptions) {
  interrupt()
  ...
}
```